drivers/bme680/bme680.c

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2019 Mesotic SAS
+ *               2020 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_bme680
+ * @{
+ * @file
+ * @brief       Bosch BME680 sensor driver implementation
+ *
+ * @author      Dylan Laduranty <dylan.laduranty@mesotic.com>
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @}
+ */
+
+#include "bme680.h"
+#include "bme680_hal.h"
+#include "bme680_params.h"
+
+#include "log.h"
+
+#ifdef MODULE_BME680_I2C
+#include "periph/i2c.h"
+#endif
+
+#ifdef MODULE_BME680_SPI
+#include "periph/spi.h"
+#endif
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+unsigned int bme680_devs_numof = 0;
+
+bme680_t *bme680_devs[BME680_NUMOF] = { };
+
+int bme680_init(bme680_t *dev, const bme680_params_t *params)
+{
+    int8_t ret;
+
+    assert(bme680_devs_numof < BME680_NUMOF);
+    assert(dev);
+
+    bme680_devs[bme680_devs_numof] = dev;
+    BME680_SENSOR(dev).dev_id = bme680_devs_numof++;
+
+    /* store interface parameters in the device for the HAL functions */
+    dev->intf = params->intf;
+
+    /* Select device interface and apply needed params */
+    if (params->ifsel == BME680_I2C_INTF) {
+#ifdef MODULE_BME680_I2C
+        BME680_SENSOR(dev).intf = BME680_I2C_INTF;
+        BME680_SENSOR(dev).read = bme680_i2c_read_hal;
+        BME680_SENSOR(dev).write = bme680_i2c_write_hal;
+#else
+        LOG_ERROR("[bme680]: module bme680_i2c not enabled\n");
+        return BME680_NO_DEV;
+#endif
+    }
+    else {
+#ifdef MODULE_BME680_SPI
+        BME680_SENSOR(dev).intf = BME680_SPI_INTF;
+        BME680_SENSOR(dev).read = bme680_spi_read_hal;
+        BME680_SENSOR(dev).write = bme680_spi_write_hal;
+        spi_init_cs(SPI_DEV(0), params->intf.spi.nss_pin);
+#else
+        LOG_ERROR("[bme680]: module bme680_spi not enabled\n");
+        return BME680_NO_DEV;
+#endif
+    }
+
+    BME680_SENSOR(dev).delay_ms = bme680_ms_sleep;
+
+    /* call internal bme680_init from Bosch Sensortech driver */
+    ret = bme680_init_internal(&BME680_SENSOR(dev));
+    if (ret != 0) {
+        DEBUG("[bme680]: Failed to get ID\n");
+        return -1;
```

Comment:
```suggestion
        return BME680_NO_DEV;
```
 But according to the return code, -1 is BME680_NULL_PTR which is weird in this situation.

Reply:
We can simply return the error code as returned  from the vendor driver.

Reply:
Error codes are simply a one-by-one mapping.

Reply:
Solved here and below.